### DISCUSSION TITLE
migrating alertmanager crds from local prometheus?

### DISCUSSION THREAD

**Discussion by clux** [grafana-cold-storage/agent#456] [Ideas]
Currently, we have a ton of `PrometheusRule` objects to control the behaviour of Alertmanager.

If we were to migrate to grafana cloud, _is there a recommended way to config manage our alerts_?

(The [grafana cloud alerting doc](https://grafana.com/docs/grafana-cloud/alerts/grafana-cloud-alerting/)  and [api docs](https://grafana.com/docs/grafana-cloud/api/) are pretty sparse.)

I feel like ideally the CRDs could be supported without having a prometheus/alertmanager installation and instead be picked up by a small _kubernetes controller_ that **reconciles against the grafana api** (perhaps embedded in one of the agents). This way people can migrate seamlessly without having to rewrite their existing alerts.

**Comment by gotjosh** [comment]
While the concept of a CRD sounds reasonable. We (at Grafana) have a preference to do it at a version control level. We've built [`cortextool`](https://github.com/grafana/cortex-tools) and a corresponding [github action](https://github.com/grafana/cortex-rules-action) as a way to interact with Grafana Cloud Rules and Alerts programmatically.

We feel that these primitives provide a good foundation for others to self-serve but as always feedback is pretty much welcome.

[Our docs](https://grafana.com/docs/grafana-cloud/alerts/alerts-rules/) cover some example interaction but again feedback is very much appreciated!

**Comment by rfratto** [comment]
Long term I'd really like to be able to support adding Prometheus rules to the Agent's config file in the same way they're defined for Prometheus. This would pair really nicely alongside reading PrometheusRule CRDs from an Operator, which we're planning on building in the next few months.

**Reply by rfratto** [reply]
I'd still like this but don't have time to implement it myself. I've opened #523 for now and labeled it as help wanted. 

(cc @gotjosh, I remember you expressing interest in having this)
